core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -141,17 +139,24 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
     }
   }
 
-  override def getListing(): Iterable[FsApplicationHistoryInfo] = applications.values
+  override def getListing(refresh: Boolean): Iterable[FsApplicationHistoryInfo] = {
+    if (refresh) checkForLogs()
```

Comment:
nit:

    if (blah) {
      code
    }

We only use the shorthand when using `if..else` as a ternary.